Userland/Libraries/LibGUI/AbstractScrollableWidget.cpp

```diff
@@ -331,11 +331,13 @@ void AbstractScrollableWidget::set_automatic_scrolling_timer_active(bool active)
 Gfx::IntPoint AbstractScrollableWidget::automatic_scroll_delta_from_position(Gfx::IntPoint pos) const
 {
     Gfx::IntPoint delta { 0, 0 };
-
+    auto autoscroll_threshold_offset = -pos.y();
     if (pos.y() < m_autoscroll_threshold)
-        delta.set_y(clamp(-(m_autoscroll_threshold - pos.y()), -m_autoscroll_threshold, 0));
-    else if (pos.y() > widget_inner_rect().height() - m_autoscroll_threshold)
-        delta.set_y(clamp(m_autoscroll_threshold - (widget_inner_rect().height() - pos.y()), 0, m_autoscroll_threshold));
+        delta.set_y(clamp(-(m_autoscroll_threshold - pos.y()), -(m_autoscroll_threshold + autoscroll_threshold_offset), 0));
+    else if (pos.y() > widget_inner_rect().height() - m_autoscroll_threshold) {
+        autoscroll_threshold_offset = pos.y() - widget_inner_rect().height();
```

Comment:
You're overwriting `autoscroll_threshold_offset` here. I suggest dropping the entire variable, you don't need it for the calculations.